### ISSUE TITLE
Migration keeps trying to add/delete a discriminator even though i specified e.HasBaseType((Type)null)

### ISSUE THREAD

**Issue by sommmen** [dotnet/efcore#33188]
Hi,

I have a couple tables setup like so:

``` csharp
public class Blog
{
    public Guid Id { get; set; }
    public long ExternalId { get; set; }
    public Guid DomainId { get; set; }
    public Domain Domain { get; set; } = null!;    
    public string? Title { get; set; }
    public DateTimeOffset CreationDateTime { get; set; }
    public DateTimeOffset ModifiedDateTime { get; set; }
    public BlogStatus Status { get; set; }
    public string? Slug { get; set; }
}

public class BlogSnapshot : Blog;
```

``` csharp
modelBuilder.Entity<Blog>(e => e.HasOne(p => p.Domain)
    .WithMany()
    .HasForeignKey(p => p.DomainId)
    .OnDelete(DeleteBehavior.Restrict));

modelBuilder.Entity<BlogSnapshot>(e =>
{
    e.HasOne(p => p.Domain)
        .WithMany()
        .HasForeignKey(p => p.DomainId)
        .OnDelete(DeleteBehavior.Restrict);

    e.HasBaseType((Type)null); // Force a separate table
    e.ToTable(t => t.IsTemporal());
});
```

I recently migrated to EF8 and now suddenly my migrations are littered with the following;
``` csharp
// Up
migrationBuilder.DropColumn(
    name: "Discriminator",
    table: "BlogSnapshots")
    .Annotation("SqlServer:IsTemporal", true)
    .Annotation("SqlServer:TemporalHistoryTableName", "BlogSnapshotsHistory")
    .Annotation("SqlServer:TemporalHistoryTableSchema", null)
    .Annotation("SqlServer:TemporalPeriodEndColumnName", "PeriodEnd")
    .Annotation("SqlServer:TemporalPeriodStartColumnName", "PeriodStart");
    
// Down
migrationBuilder.AddColumn<string>(
    name: "Discriminator",
    table: "BlogSnapshots",
    type: "nvarchar(max)",
    nullable: false,
    defaultValue: "")
    .Annotation("SqlServer:IsTemporal", true)
    .Annotation("SqlServer:TemporalHistoryTableName", "BlogSnapshotsHistory")
    .Annotation("SqlServer:TemporalHistoryTableSchema", null)
    .Annotation("SqlServer:TemporalPeriodEndColumnName", "PeriodEnd")
    .Annotation("SqlServer:TemporalPeriodStartColumnName", "PeriodStart");
``` 

For some reason it keeps on trying to create this discriminator column.

I'm not sure whether this could be a bug, or i'm doing something dumb.

If i run the migration command the `ModelSnapshot` is not being updated (no changes saids git), i can see:
``` csharp
b.ToTable("BlogSnapshots");

b.HasDiscriminator().HasValue("BlogSnapshot");

b.ToTable(tb => tb.IsTemporal(ttb =>
        {
            ttb.UseHistoryTable("BlogSnapshotsHistory");
            ttb
                .HasPeriodStart("PeriodStart")
                .HasColumnName("PeriodStart");
            ttb
                .HasPeriodEnd("PeriodEnd")
                .HasColumnName("PeriodEnd");
        }));
```

In both the <migration>.Designer.cs file and the model snapshot. Running multiple empty migrations will generate the same changes where it tries to delete the discriminator.

This was not the behavior i had in EF7.

Let me know if i'm either doing something unsupported or if this could indeed be a bug.

 <details>
  <summary>Why i'm doing things this way</summary>
  
  Temporal tables track changes, but in my case i'm using a [bulk update library ](https://github.com/borisdj/EFCore.BulkExtensions) which generates a lot of `UPDATE` statements. This would cause a lot of rows in the history table when the data did not really change.
  
So i have a main table and sync this with the 'SnapShots' table via a query and that way my temporal data only contains actual changes. 
</details> 

### Include provider and version information

EF Core version: 8.0.2
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 8.0.2
Operating system: Win11
IDE: Latest vs

**Comment by ajcvickers** [comment]
Note for team triage: still repros on latest daily; regression.

```C#
using (var context = new SomeDbContext())
{
}

public class SomeDbContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        => optionsBuilder
            .UseSqlServer(@"Data Source=localhost;Database=One;Integrated Security=True;Trust Server Certificate=True")
            .LogTo(Console.WriteLine, LogLevel.Information)
            .EnableSensitiveDataLogging();
	
    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.Entity<Blog>(e => e.HasOne(p => p.Domain)
            .WithMany()
            .HasForeignKey(p => p.DomainId)
            .OnDelete(DeleteBehavior.Restrict));

        modelBuilder.Entity<BlogSnapshot>(e =>
        {
            e.HasOne(p => p.Domain)
                .WithMany()
                .HasForeignKey(p => p.DomainId)
                .OnDelete(DeleteBehavior.Restrict);

            e.HasBaseType((Type)null); // Force a separate table
            e.ToTable(t => t.IsTemporal());
        });
    }
}

public class Blog
{
    public Guid Id { get; set; }
    public long ExternalId { get; set; }
    public Guid DomainId { get; set; }
    public Domain Domain { get; set; } = null!;
    public string? Title { get; set; }
    public DateTimeOffset CreationDateTime { get; set; }
    public DateTimeOffset ModifiedDateTime { get; set; }
    public string? Slug { get; set; }
}

public class Domain
{
    public Guid Id { get; set; }
}

public class BlogSnapshot : Blog;
```

**Comment by ajcvickers** [comment]
@maumar #32763, #31330, and #31314 are similar issues.

**Comment by sommmen** [comment]
This seems to have been solved somewhere in the minor versions, as i'm no longer getting this issue.

**Comment by ajcvickers** [comment]
Note for team: I think this one slipped off our radar when looking at 8.0 regressions. It was closed by a customer, and I verified that this repro is indeed fixed in the work for #33605.
